### ISSUE TITLE
[firebase_auth] equivalent to admin.auth().generateEmailVerificationLink()

### ISSUE THREAD

**Issue by logemann** [FirebaseExtended/flutterfire#1383]
**Describe the bug**
admin.auth().generatePasswordResetLink() is a method from the node.js admin firebase package. There seems no equivalent in the dart world with Firebase auth. Same goes for generatePasswordResetLink() ? 

**Expected behavior**
API for getting the email link and ResetLink

To elaborate further on this one: The goal is to send a custom email verficiation email which looks better than what firebase provides but for that i need the URL which firebase would generate for their very own verify-email. Dont know if generateEmailVerificationLink() even does that...

**Comment by logemann** [comment]
perhaps its not a bug but a featRequest?

**Comment by iapicca** [comment]
@logemann  you read my mind
thank you for your contribution

**Comment by logemann** [comment]
Tested the node.js version (in fact i used typesript but anyway) on cloud function on auth.createUser() event and the following worked perfect:

let verifyLink = await auth.generateEmailVerificationLink(user.email!);

this got me the firebase URL for link verification. I used that for my custom email i sent via Sendgrid. But still would like to see that in the dart package. Perhaps the  generatePasswordResetLink() is even more important beacause this cant be done easily in the backend cause the trigger will be the frontend. Of course i can circumvent this by calling a custom function on the backend but still a workaround.
